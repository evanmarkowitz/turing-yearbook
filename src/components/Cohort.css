.cohort--card--container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.cohort__title {
  background-color: greenyellow;
  padding: 10px;
  color: 	#d46ce7;
  font-size: 36px;
}