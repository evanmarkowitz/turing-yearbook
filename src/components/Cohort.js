import React from 'react';
import Person from './Person';
import './Cohort.css';

const Cohort = props => {
  const showPerson = props.data.map(pers => {
      return <Person photo = {pers.photo} name = {pers.name} quote={pers.quote} superlative = {pers.superlative} />
    })
  return (
    <div className='cohort'>
      <h2 className="cohort__title">{props.title}</h2>
      <div className= "cohort--card--container">
        {showPerson}
      </div>
    </div>)
}

export default Cohort;
