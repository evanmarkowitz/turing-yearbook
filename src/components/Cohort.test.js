import React from 'react'; 
import { shallow, mount } from 'enzyme';
import Cohort from './Cohort.js';

describe('Person', () => { 
  let testData = []
  beforeEach(() => {
    testData = [{id: 2, name: 'Amy', quote: 'I believe in you, cuties!', superlative: 'Most Likely to Be an Actual Superhero', photo: './amy.jpg'},]
  })
  it('should match the snapshot with all data passed in correctly', () => { 
    const wrapper = mount( <Cohort
      data= {testData}
      />);
    expect(wrapper).toMatchSnapshot(); 
  }); 
  
}); 