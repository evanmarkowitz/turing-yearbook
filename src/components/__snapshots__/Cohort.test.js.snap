// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should match the snapshot with all data passed in correctly 1`] = `
<Cohort
  data={
    Array [
      Object {
        "id": 2,
        "name": "Amy",
        "photo": "./amy.jpg",
        "quote": "I believe in you, cuties!",
        "superlative": "Most Likely to Be an Actual Superhero",
      },
    ]
  }
>
  <div
    className="cohort"
  >
    <h2
      className="cohort__title"
    />
    <div
      className="cohort--card--container"
    >
      <Person
        name="Amy"
        photo="./amy.jpg"
        quote="I believe in you, cuties!"
        superlative="Most Likely to Be an Actual Superhero"
      >
        <section
          className="person--card"
        >
          <img
            alt="portrait"
            src="./amy.jpg"
          />
          <h3
            className="person__name"
          >
            Amy
          </h3>
          <p
            className="person__quote"
          >
            I believe in you, cuties!
          </p>
          <p
            className="person__superlative"
          >
            Most Likely to Be an Actual Superhero
          </p>
        </section>
      </Person>
    </div>
  </div>
</Cohort>
`;
